test/test_field.jl

```diff
@@ -451,4 +450,65 @@ end
+
+            CUDA.@allowscalar begin
+                # First test that the regular view is correct
+                cv = view(c, :, :, 1+1:k_top-1)
+                # cv = view(c, :, :, 2:7) for a ZFaceField with bounded topology in z else cv = view(c, :, :, 2:6).
+                @test size(cv) == (1, 1, k_top-2)
+                @test size(parent(cv)) == (1+2Hx, 1+2Hy, k_top-2)
+                @test all(cv[1, 1, i] == c[1, 1, i] for i in 1+1:k_top-1)
+
+                # Now test the views of views
+                cvv = view(cv, :, :, 1+2:k_top-2)
+                # cvv = view(cv, :, :, 3:6) for a ZFaceField with bounded topology in z else cvv = view(cv, :, :, 3:5).
+                @test size(cvv) == (1, 1, k_top-4)
+                @test size(parent(cvv)) == (1+2Hx, 1+2Hy, k_top-4)
+                @test all(cvv[1, 1, i] == cv[1, 1, i] for i in 1+2:k_top-2)
+
+                cvvv = view(cvv, :, :, 1+3:k_top-3)
+                # cvvv = view(cvv, :, :, 4:5) for a ZFaceField with bounded topology in z else cvvv = view(cvv, :, :, 4:4).
+                @test size(cvvv) == (1, 1, k_top-6)
+                @test size(parent(cvvv)) == (1+2Hx, 1+2Hy, k_top-6)
+                @test all(cvvv[1, 1, i] == cvv[1, 1, i] for i in 1+3:k_top-3)
+
+                @test_throws ArgumentError view(cv, :, :, 1)
+                @test_throws ArgumentError view(cv, :, :, k_top)
+                # @test_throws ArgumentError view(cv, :, :, 8) for a ZFaceField with bounded topology in z else @test_throws ArgumentError view(cv, :, :, 7).
+                @test_throws ArgumentError view(cvv, :, :, 1:1+1)
+                @test_throws ArgumentError view(cvv, :, :, k_top-1:k_top)
+                # @test_throws ArgumentError view(cvv, :, :, 6:7) for a ZFaceField with bounded topology in z else @test_throws ArgumentError view(cvv, :, :, 5:6).
+                @test_throws ArgumentError view(cvvv, :, :, 1:1+2)
+                @test_throws ArgumentError view(cvvv, :, :, k_top-2:k_top)
+                # @test_throws ArgumentError view(cvvv, :, :, 6:8) for a ZFaceField with bounded topology in z else @test_throws ArgumentError view(cvvv, :, :, 5:7).
+
+                @test_throws BoundsError cv[:, :, 1]
+                @test_throws BoundsError cv[:, :, k_top]
+                # @test_throws BoundsError cv[:, :, 8] for a ZFaceField with bounded topology in z else @test_throws BoundsError cv[:, :, 7].
+                @test_throws BoundsError cvv[:, :, 1:1+1]
+                @test_throws BoundsError cvv[:, :, k_top-1:k_top]
+                # @test_throws BoundsError cvv[:, :, 7:8] for a ZFaceField with bounded topology in z else @test_throws BoundsError cvv[:, :, 6:7].
+                @test_throws BoundsError cvvv[:, :, 1:1+2]
+                @test_throws BoundsError cvvv[:, :, k_top-2:k_top]
+                # @test_throws BoundsError cvvv[:, :, 6:8] for a ZFaceField with bounded topology in z else @test_throws BoundsError cvvv[:, :, 5:7].

```

Comment:
@siddharthabishnu can you put `@allowscalar` around only the specific lines where it is needed? I thought by using `k_top` we eliminated the need for the comments. Not sure what the comments mean

Reply:
@glwagner, agreed. Resolved in commit [427cb51](https://github.com/CliMA/Oceananigans.jl/pull/3573/commits/427cb51b0a4abef57dd6f504bfa1c5fef0096eed).